kernel: bump 6.6 to 6.6.38

graysky2 opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.38

Removed `target/linux/generic/hack-6.6/900-fix-build-to-handle-return-value.patch` which was introduced as a hacky attempt to fix an upstream issue related to 6.6.37.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

graysky2 committed:
kernel: bump 6.6 to 6.6.37

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.37

Added:
	generic/hack-6.6/900-fix-build-to-handle-return-value.patch[1]

Manually rebased:
	generic/pending-6.6/834-ledtrig-libata.patch

Removed upstreamed:
	bcm27xx/patches-6.6/950-0398-drm-panel-panel-ilitek9881c-Use-cansleep-methods.patch[2]

All other patches automatically rebased.

1. Patch suggested by @DragonBluep to circumvent upstream breakage of kernel 6.6.37 compilation. See comments in https://github.com/openwrt/openwrt/pull/15879 for additional discussion.
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.37&id=1618f7a875ffd916596392fd29880c0429b8af60

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/15879
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

graysky2 commented:
Something in this bump breaks flogic building:
```
% make -j1 V=s
...
make -C /scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.6.37 KCFLAGS="-fmacro-prefix-map=/scratch/union/build_dir/target-aarch64_cortex-a53_musl=target-aarch64_cortex-a53_musl -mcpu=cortex-a53 -mcpu=cortex-a53" HOSTCFLAGS="-O2 -I/scratch/union/staging_dir/host/include  -Wall -Wmissing-prototypes -Wstrict-prototypes" CROSS_COMPILE="aarch64-openwrt-linux-musl-" ARCH="arm64" KBUILD_HAVE_NLS=no KBUILD_BUILD_USER="" KBUILD_BUILD_HOST="" KBUILD_BUILD_TIMESTAMP="Fri Jul  5 10:07:11 2024" KBUILD_BUILD_VERSION="0" KBUILD_HOSTLDFLAGS="-L/scratch/union/staging_dir/host/lib" CONFIG_SHELL="bash" V=''  cmd_syscalls=  CC="aarch64-openwrt-linux-musl-gcc" KERNELRELEASE=6.6.37 Image dtbs modules
make[5]: Entering directory '/scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.6.37'
  CC      arch/arm64/net/bpf_jit_comp.o
arch/arm64/net/bpf_jit_comp.c: In function 'bpf_int_jit_compile':
arch/arm64/net/bpf_jit_comp.c:1651:17: error: ignoring return value of 'bpf_jit_binary_lock_ro' declared with attribute 'warn_unused_result' [-Werror=unused-result]
 1651 |                 bpf_jit_binary_lock_ro(header);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[9]: *** [scripts/Makefile.build:243: arch/arm64/net/bpf_jit_comp.o] Error 1
make[8]: *** [scripts/Makefile.build:480: arch/arm64/net] Error 2
make[7]: *** [scripts/Makefile.build:480: arch/arm64] Error 2
make[6]: *** [/scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.6.37/Makefile:1921: .] Error 2
make[5]: *** [Makefile:234: __sub-make] Error 2
make[5]: Leaving directory '/scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.6.37'
make[4]: *** [Makefile:18: /scratch/union/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.6.37/.modules] Error 2
make[4]: Leaving directory '/scratch/union/target/linux/mediatek'
make[3]: *** [Makefile:11: compile] Error 2
make[3]: Leaving directory '/scratch/union/target/linux'
time: target/linux/compile#1.76#0.85#2.67
    ERROR: target/linux failed to build.
make[2]: *** [target/Makefile:32: target/linux/compile] Error 1
make[2]: Leaving directory '/scratch/union'
make[1]: *** [target/Makefile:25: /scratch/union/staging_dir/target-aarch64_cortex-a53_musl/stamp/.target_compile] Error 2
make[1]: Leaving directory '/scratch/union'
make: *** [/scratch/union/include/toplevel.mk:248: world] Error 2
```

DragonBluep commented:
It's an upstream issue. I guess this patch set can fix it. You can find it on torvalds' repo https://github.com/torvalds/linux.
https://lore.kernel.org/all/20240228141824.119877-1-puranjay12@gmail.com/

Or you can add a new patch to handle the return value.
```
+int ret;
-bpf_jit_binary_lock_ro(header);
+ret = bpf_jit_binary_lock_ro(header);
```

namiltd commented:
https://www.spinics.net/lists/stable/msg755793.html

DragonBluep commented on target/linux/generic/hack-6.6/900-fix-build-to-handle-return-value.patch:
```diff
@@ -0,0 +1,12 @@
+ 			goto out_off;
+ 		}
+-		bpf_jit_binary_lock_ro(header);
++		int ret;
```
The variable definition should be placed at the beginning of the function. Although it is already a new feature of the modern C language.

graysky2 replied:
I'm sorry, I do not know what you're asking me to do.  Could you please provide a diff of the newly created `900-fix-build-to-handle-return-value.patch` for me to use to modify to your liking?

namiltd replied:
```
--- a/arch/arm64/net/bpf_jit_comp.c
+++ b/arch/arm64/net/bpf_jit_comp.c
@@ -1526,6 +1526,7 @@
 	bool extra_pass = false;
 	struct jit_ctx ctx;
 	u8 *image_ptr;
+	int ret;
 
 	if (!prog->jit_requested)
 		return orig_prog;
@@ -1648,7 +1649,7 @@
 			prog->jited_len = 0;
 			goto out_off;
 		}
-		bpf_jit_binary_lock_ro(header);
+		ret = bpf_jit_binary_lock_ro(header);
 	} else {
 		jit_data->ctx = ctx;
 		jit_data->image = image_ptr;

graysky2 replied:
Thanks @namiltd

namiltd commented:
```
+--- a/arch/arm64/net/bpf_jit_comp.c
++++ b/arch/arm64/net/bpf_jit_comp.c
+@@ -1526,6 +1526,7 @@
+ 	bool extra_pass = false;
+ 	struct jit_ctx ctx;
+ 	u8 *image_ptr;
++	int ret;
+ 
+ 	if (!prog->jit_requested)
+ 		return orig_prog;
+@@ -1648,7 +1649,7 @@
+ 			prog->jited_len = 0;
+ 			goto out_off;
+ 		}
+-		bpf_jit_binary_lock_ro(header);
++		ret = bpf_jit_binary_lock_ro(header);
+ 	} else {
+ 		jit_data->ctx = ctx;
+ 		jit_data->image = image_ptr;

graysky2 commented:
I verified a new build from https://github.com/openwrt/openwrt/commit/928791118cdc770fdeaa0ecb8fa67ffd3026f982 works on flogic/xiaomi_redmi-router-ax6000-ubootmod

hauke commented:
> It's an upstream issue. I guess this patch set can fix it. You can find it on torvalds' repo https://github.com/torvalds/linux. https://lore.kernel.org/all/20240228141824.119877-1-puranjay12@gmail.com/
> 
> Or you can add a new patch to handle the return value.
> 
> ```
> +int ret;
> -bpf_jit_binary_lock_ro(header);
> +ret = bpf_jit_binary_lock_ro(header);
> ```

This is the upstream discussion:
https://lore.kernel.org/netdev/5A29E00D83AB84E3+20240706031101.637601-1-wangyuli@uniontech.com/T/#m84d3f365d2b9993e757d05f1ea779631fa3a4982

vincejv commented:
6.6.38 just got released, might want to rebase...

graysky2 committed:
kernel: bump 6.6 to 6.6.38

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.38

Removed target/linux/generic/hack-6.6/900-fix-build-to-handle-return-value.patch
which was introduced as a hacky attempt to fix an upstream issue related to 6.6.37.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/15879
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

hauke commented:
6.6.38 reverts the patches related to `bpf_jit_binary_lock_ro()`. In the mail thread people also said that powerpc was crashing after loading the first bpf program. Please switch to 6.6.38.
